### ISSUE TITLE
Allow setting favicon in a theme

### ISSUE THREAD

**Issue by bbranan** [DSpace/dspace-angular#1347]
**Is your feature request related to a problem? Please describe.**

I'd like to be able to specify a custom favicon image from within a theme. This image would override the default DSpace favicon.

**Describe the solution you'd like**

I expect there would be two steps to this:
1. Place the custom favicon.ico image in the /images directory of the active theme
2. Specify the path to the new favicon image in a way that it replaces the default favicon path

Step 2 could be omitted if an automated step were added which looked for a file named favicon.ico in the active theme and either replaced the default favicon image or updated the favicon path to point to this image.

**Describe alternatives or workarounds you've considered**

The current work-around is to place the favicon image in the theme, then update the top-level index.html value to change:
```
<link rel="icon" type="image/x-icon" href="assets/images/favicon.ico" />
```
to:
```
<link rel="icon" type="image/x-icon" href="/assets/<my-theme>/images/favicon.ico" />
```

This method is effective, it simply breaks the theme model by requiring changes to the look of a site to be made outside of the active theme.

**Additional context**

**Comment by tdonohue** [comment]
@artlowel : Per today's meeting, assigning you/your team to work on and adding your 4hr estimate.

**Comment by artlowel** [comment]
@tdonohue I've looked into this in more detail now, and I'd like to change my estimate to **8 hours**. The reason is that instead of just adding support for a single favicon.ico file, I'd like to include more high res icons for iOS and Android, or SVG favicons for browsers that support it. This means adding multiple configurable tags per theme to the `<head>` instead of just one hardcoded one. I'd also like to apply the `extends` mechanism for themes we added in 7.1 to favicons as well.
